controller/appcontroller.go

```diff
@@ -1195,6 +1195,14 @@ func (ctrl *ApplicationController) processRequestedAppOperation(app *appv1.Appli
 				ctrl.requestAppRefresh(app.QualifiedName(), CompareWithLatest.Pointer(), &retryAfter)
 				return
 			} else {
+				if state.Phase == synccommon.OperationRunning && app.Spec.SyncPolicy.Retry.Refresh && app.Status.Sync.Revision != state.Operation.Sync.Revision {
+					logCtx.Infof("A new revision is available, refreshing and terminating app, was phase: %s, message: %s", state.Phase, state.Message)
+					ctrl.requestAppRefresh(app.QualifiedName(), CompareWithLatest.Pointer(), nil)
+					state.Phase = synccommon.OperationTerminating
+					state.Message = "Operation forced to terminate (new revision available)"
```

Comment:
I could not find this message on the UI, is there a way to make it appear?

Reply:
I think we have to move `setOperationState` above the `requestAppRefresh`, since otherwise there can be a race condition by the refresh processor kicking in and message not being ready yet.